internal/cache/controller_obj_cache.go

```diff
@@ -148,17 +174,25 @@ func (c *AviObjCache) PopulateVsMetaCache() {
 			vs_cache_obj, foundvs := vsObj.(*AviVsCache)
 			if foundvs {
 				vs_cache_obj.ReplaceSNIChildCollection(sniChildUuids)
+
+				if isEVHEnabled {
+					childNSNameToDelete, childUuidToDelete = c.listEVHChildrenToDelete(vs_cache_obj, sniChildUuids)
```

Comment:
For multiple parentsVSes, childUuidToDelete would contain children of last parent VS only. This would be a problem. 
Also, do we need to handle the upgrade case, as evh was in tech preview mode only ? @sudswasavi

Reply:
@monotosh-avi : We can deprecate this function after 1-2 releases. But if we do not delete existing non-encoded EVH Vs then it would not allow to create new encoded VS (As encoded VS will have same host, path). 
Regarding comment of childUuidToDelete, children VSes will be associated with Shared Vs. These child VS will have name without encoding. So we will delete them all with this. I am not getting problem here.

Reply:
I was thinking of a case where there are 8 parents VSes  (vs-1 to vs-8 ) each with 10 child VSes .. so we have to delete all 80 children.
As we start going through all the parents, we would add the children .. so after going though vs-1 we would add 8 children to childUuidToDelete. But once we go through vs-2 we would populate childUuidToDelete with the children of vs-2 and replace children of vs-1. I think we need to append the children.